src/ports/validation/component.ts

```diff
@@ -0,0 +1,34 @@
+import 'isomorphic-fetch'
+import Ajv from 'ajv'
+import { AppComponents } from '../../types'
+import { IValidationComponent, Schema } from './types'
+
+export async function createValidationComponent(
+  components: Pick<AppComponents, 'logs'>
+): Promise<IValidationComponent> {
+  const { logs } = components
+  const logger = logs.getLogger('validations-component')
+
+  const ajv = new Ajv()
+
+  // Methods
+  function require<T>(schema: Schema<T>, data: any) {
+    logger.debug('Requiring object with schema', schema)
+    const validator = ajv.compile<T>(schema)
+    validator(data)
+    if (validator.errors) {
+      throw new Error(`Invalid data: ${JSON.stringify(validator.errors)}`)
+    }
+  }
+
+  function validate<T>(schema: Schema<T>, data: any): data is T {
+    // TODO: Should we make errors available in this case?
+    logger.debug('Validating object with schema', schema)
+    return ajv.validate<T>(schema, data)
+  }
+
+  return {
+    require,
+    validate,
+  }
```

Comment:
Even though this works as a component, it is also true that those are pure functions and can belong to the "logic" area of the hexagonal architecture. Enabling us to just export the functions at top level of the file

Reply:
Awesome! I wasn't sure about this one really, I ended up creating a component mostly to be able to reuse it. I'll move it to logic, it makes much more sense